src/shared_state.h

```diff
@@ -41,6 +41,8 @@ struct RapidsSharedState : rapids::SharedModelState {
     predict_proba_ = get_config_param<bool>("predict_proba", false);
     model_format_ = string_to_serialization(
         get_config_param<std::string>("model_type", std::string{"xgboost"}));
+    xgboost_allow_unknown_field_ =
```

Comment:
Do you think we should default to true or false here? Asking because I am honestly unsure myself and could argue it either way.

Reply:
I vote for defaulting to false, because unknown fields can help us find incompatibilities with the latest XGBoost model format.

Reply:
Works by me! I'll merge this as soon as it passes CI.